### DISCUSSION TITLE
[N64] Issue Dumping 2 Japanese Titles

### DISCUSSION THREAD

**Discussion by majorpbx** [sanni/cartreader#328] [Need help]
I came across two carts that for whatever reason will not dump with v7.3. Explanation below.

Carts:
1. [NUS-NGPJ-JPN] Goemon - Mononoke Sugoroku
![IMG_2214](https://user-images.githubusercontent.com/18235905/153634259-703a133c-7e17-4121-9c5b-c4170de82960.jpg)

2. [NUS-NMHJ-JPN] Mahjong Hourouki Classic
![IMG_2215](https://user-images.githubusercontent.com/18235905/153634341-72e3a5af-e4ba-40d6-bb97-427cb837afc9.jpg)

Front Covers:
![IMG_2211](https://user-images.githubusercontent.com/18235905/153633143-ea5351f6-f090-45c7-9839-1c9ee8cbf6cf.jpg)

Cart Info Screen:
![IMG_2212](https://user-images.githubusercontent.com/18235905/153633401-f7754dc5-ebe5-4610-9132-4166015be206.jpg)
![IMG_2213](https://user-images.githubusercontent.com/18235905/153633419-475a23ff-cc13-478c-b207-16851411140d.jpg)

n64.txt data:
965c4575,NGPJ,16,0
990a8e54,NMHJ,12,0

Issue:
Once you reach the "Read ROM" screen, I select that option and the Cart Reader jumps back to the main menu.

**Comment by sanni** [comment]
Could be some character in the name the Cart Reader can't write. 

I'm excluding characters that make it crash here: https://github.com/sanni/cartreader/blob/master/Cart_Reader/N64.ino#L2406
These must be new ones.

**Reply by majorpbx** [reply]
I looked at that code and figured just as you said. I was looking into supporting UTF8 or Japanese character set for the SD, Arduino, etc... But just wanted everyone to be aware of the issue.

**Reply by majorpbx** [reply]
Just found rom dumps online so have no idea if these are accurate but here is the ROM Name Fields.

[NUS-NGPJ-JPN] Goemon - Mononoke Sugoroku
`Big-Endian
0x20 - BADEB4D3DDD3C9C9B9BDBADEDBB8202020202020`

[NUS-NMHJ-JPN] Mahjong Hourouki Classic
`Big-Endian
0x20 - CFB0BCDEACDDCEB3DBB3B7434C41535349432020`

**Comment by sanni** [comment]
I can't replicate the error. I could only find out that it has nothing to do with any special chars, so my initial guess was wrong.

![IMG_1069](https://user-images.githubusercontent.com/19169494/153718994-95c2fedd-df5a-48af-b5c0-f2c2150c183e.jpg)

Have you tried with an empty SD with just the SD files and with the folder number reset. For resetting the folder number hold and keep holding the left button while hovering over about until it says "Resetting folder". You have to keep holding it for quite a while even after it enters the about screen.
Just to make sure it's not some file or folder count limit that has been reached.

**Reply by majorpbx** [reply]
I was able to get [NUS-NGPJ-JPN] Goemon - Mononoke Sugoroku to dump successfully this way. Tried the same method (without resetting folder EEPROM function) and [NUS-NMHJ-JPN] Mahjong Hourouki Classic still skips to the main menu. I'll keep trying to see if I can get it to work. Odd that I've received the error that the header is corrupt a few times but that may just be due to the slot I'm using. Meaning I have to be careful about my carts seating in the slot.

**Comment by sanni** [comment]
You can step through the code with some added wait(); and then press a button to get to the next wait(); to see where exactly it resets:

````
/ Read rom and save to the SD card
void readRom_N64() {

display_Clear();
println_Msg(F("FILE NAME"));
display_Update();
wait();

  // Get name, add extension and convert to char array for sd lib
  strcpy(fileName, romName);
  strcat(fileName, ".Z64");


display_Clear();
println_Msg(F("FOLDER NAME"));
display_Update();
wait();

redumpnewfolder:
  // create a new folder
  EEPROM_readAnything(0, foldern);
  sprintf(folder, "N64/ROM/%s/%d", romName, foldern);
  sd.mkdir(folder, true);
  sd.chdir(folder);

  display_Clear();
  print_Msg(F("Saving to "));
  print_Msg(folder);
  println_Msg(F("/..."));
  display_Update();

  // write new folder number back to eeprom
  foldern = foldern + 1;
  EEPROM_writeAnything(0, foldern);

display_Clear();
println_Msg(F("OPEN SD"));
display_Update();
wait();

redumpsamefolder:
  // Open file on sd card
  if (!myFile.open(fileName, O_RDWR | O_CREAT)) {
    print_Error(F("SD Error"), true);
  }
````

**Reply by majorpbx** [reply]
I have no idea why but just inserting those pauses did the trick. It dumped beautifully first try.

**Comment by sanni** [comment]
Interesting. Can you remove the pauses and only add one at a time to see what pause exactly is the one that fixes it?
If we identify where the code is too fast we could add a delay(100); there maybe this would fix it.

**Reply by majorpbx** [reply]
Tried that and had to add all three breaks back in. Now all i get is this after that first good dump. Cart issue maybe? This has me stumped.

![3857141A-106E-484B-A6C7-193EBF6A5135](https://user-images.githubusercontent.com/18235905/154751216-e462002e-548d-4396-8b5b-6159721e438d.jpeg)

**Reply by majorpbx** [reply]
Just odd.

Dump details:

Name	: CLASSIC
ID	: NMHJ
ROM CRC1: x�j{
ROM CRC2: j{ %�d
Size	: 12MB
Save	: unknown
Version	: 1.0
Saved To: N64/ROM/CLASSIC/0
CRC	: 990a8e54 [Match]
Time	: 71

My details:

------- Cart Info -----------------------------------
Name    : ﾏｰｼﾞｬﾝﾎｳﾛｳｷCLASSIC
ByteOrd : Big-Endian
ID      : NMHJ
Version : 1.0
CRC1    : CCCC821E
CRC2    : 96E88A83
CIC Chip: CIC-NUS-6102/7101
